server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -220,19 +224,35 @@ static ClusterState addIndexClosedBlocks(final Index[] indices, final ClusterSta
 
         for (IndexMetaData indexToClose : indicesToClose) {
             final Index index = indexToClose.getIndex();
-            if (currentState.blocks().hasIndexBlock(index.getName(), INDEX_CLOSED_BLOCK) == false) {
-                blocks.addIndexBlock(index.getName(), INDEX_CLOSED_BLOCK);
+
+            ClusterBlock indexBlock = null;
+            final Set<ClusterBlock> clusterBlocks = currentState.blocks().indices().get(index.getName());
+            if (clusterBlocks != null) {
+                for (ClusterBlock clusterBlock : clusterBlocks) {
+                    if (clusterBlock.id() == INDEX_CLOSED_BLOCK_ID) {
+                        // Reuse the existing index closed block
+                        indexBlock = clusterBlock;
+                        break;
+                    }
+                }
             }
             if (useDirectClose) {
                 logger.debug("closing index {} directly", index);
-                metadata.put(IndexMetaData.builder(indexToClose).state(IndexMetaData.State.CLOSE));
+                metadata.put(IndexMetaData.builder(indexToClose).state(IndexMetaData.State.CLOSE)); // increment version?
+                blocks.removeIndexBlockWithId(index.getName(), INDEX_CLOSED_BLOCK_ID);
                 routingTable.remove(index.getName());
+                indexBlock = INDEX_CLOSED_BLOCK;
             }
-            blockedIndices.add(index);
+            if (indexBlock == null) {
```

Comment:
put this in an `else` to the `useDirectClose` and assert that indexBlock has a UUID